private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts

```diff
@@ -717,7 +719,22 @@ export class TemplateServiceClient {
         this._incrementTemplateHits(templateId, templates![0], version);
       }
       sortTemplateByVersion(templates![0]);
-      if (!version) return { success: true, result: templates };
+      if (!version) {
+        if (data) {
+          let resultTemplates: ITemplate[] = [];
+          let template: ITemplate = templates[0];
+          if (!template.instances || !template.instances[0]) {
+            return { success: false, errorMessage: "No template instances found" };
+          }
+          let boundJSON: JSON = createCard(template.instances[0].json, data);
+          template.instances![0].json = boundJSON;
+          //return only the most recent version of the template with data bound 
```

Comment:
If hit the endpoint specifying a version, will it bind that version or the most recent version

Reply:
It will bind that version, not the most recent version.